include/pybind11/numpy.h

```diff
@@ -1986,7 +1990,7 @@ struct vectorize_helper {
         // Pointers to values the function was called with; the vectorized ones set here will start
         // out as array_t<T> pointers, but they will be changed them to T pointers before we make
         // call the wrapped function.  Non-vectorized pointers are left as-is.
-        std::array<void *, N> params{{&args...}};
+        std::array<void *, N> params{{reinterpret_cast<void *>(&args)...}};
```

Comment:
It seems like most uses of `reinterpret_cast<void*>(&...)` in this cl should be changed to `static_cast<void*>(&...)` instead.

Reply:
> The best way to review here is to start with the "Files changed" tab near the top of this PR.

Oh, sorry, I just now realized it doesn't look the way I was expecting. Ugh!

Reply:
Here is a helper PR showing the net diff (what I was wrongly hoping "Files changed" would show):

https://github.com/pybind/pybind11/pull/5277/files

Reply:
Ok. So perhaps make this change at head. :)